src/engine/Account-Imap.h

```diff
@@ -0,0 +1,153 @@
+/********************************************************************\
+ * This program is free software; you can redistribute it and/or    *
+ * modify it under the terms of the GNU General Public License as   *
+ * published by the Free Software Foundation; either version 2 of   *
+ * the License, or (at your option) any later version.              *
+ *                                                                  *
+ * This program is distributed in the hope that it will be useful,  *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of   *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the    *
+ * GNU General Public License for more details.                     *
+ *                                                                  *
+ * You should have received a copy of the GNU General Public License*
+ * along with this program; if not, contact:                        *
+ *                                                                  *
+ * Free Software Foundation           Voice:  +1-617-542-5942       *
+ * 51 Franklin Street, Fifth Floor    Fax:    +1-617-542-2652       *
+ * Boston, MA  02110-1301,  USA       gnu@gnu.org                   *
+ *                                                                  *
+\********************************************************************/
+/** @addtogroup Engine
+    @{ */
+/** @addtogroup Account
+    Account Import mapping functions
+    @{ */
+/** @file Account-Imap.h
+    @brief Account Imap handling public routines
+    @author Copyright (C) 2007 David Hampton
+    @author Copyright (C) 2016 Robert Fewell
+*/
+
+#ifndef ACCOUNT_IMAP_H
+#define ACCOUNT_IMAP_H
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include "qof.h"
+#include "gnc-engine.h"
+#include "policy.h"
+
+typedef struct imap_info
+{
+    Account        *source_account;
+    Account        *map_account;
+    GList          *list;
+    char           *category_head;
+    char           *full_category;
+    char           *match_string;
+    char           *count;
+}GncImapInfo;
```

Comment:
This doesn't seem to be exported to anything, so it should move to Accounts_imap.cpp. That will allow you to replace the char*s with std::string, which will clean up your code quite a bit since std::strings take care of their own allocations.